src/ledger/cleanup_service.zig

```diff
@@ -401,53 +402,99 @@ const TestDB = ledger.tests.TestDB;
+    // This ensures that the memtable is flushed to disk, which is needed in other
+    // to retrive live files, in the rest of the test.
+    {
+        var db = try TestDB.init(@src());
+        defer db.deinit();
 
-    var lowest_cleanup_slot = sig.sync.RwMux(Slot).init(0);
-    var max_root = std.atomic.Value(Slot).init(0);
+        var lowest_cleanup_slot = sig.sync.RwMux(Slot).init(0);
+        var max_root = std.atomic.Value(Slot).init(0);
 
-    var reader = try BlockstoreReader.init(
-        allocator,
-        logger,
-        db,
-        registry,
-        &lowest_cleanup_slot,
-        &max_root,
-    );
+        var reader = try BlockstoreReader.init(
+            allocator,
+            logger,
+            db,
+            registry,
+            &lowest_cleanup_slot,
+            &max_root,
+        );
 
-    // set highest and lowest slot by inserting slot_meta
-    var lowest_slot_meta = ledger.meta.SlotMeta.init(allocator, 10, null);
-    defer lowest_slot_meta.deinit();
-    lowest_slot_meta.received = 10;
+        // set highest and lowest slot by inserting slot_meta
+        var lowest_slot_meta = ledger.meta.SlotMeta.init(allocator, 10, null);
+        defer lowest_slot_meta.deinit();
+        lowest_slot_meta.received = 10;
+
+        var highest_slot_meta = ledger.meta.SlotMeta.init(allocator, 20, null);
+        defer highest_slot_meta.deinit();
+        highest_slot_meta.received = 20;
+
+        {
+            var write_batch = try db.initWriteBatch();
+            defer write_batch.deinit();
+            try write_batch.put(
+                ledger.schema.schema.slot_meta,
+                lowest_slot_meta.slot,
+                lowest_slot_meta,
+            );
+            try write_batch.put(
+                ledger.schema.schema.slot_meta,
+                highest_slot_meta.slot,
+                highest_slot_meta,
+            );
+            try db.commit(&write_batch);
+        }
 
-    var highest_slot_meta = ledger.meta.SlotMeta.init(allocator, 20, null);
-    defer highest_slot_meta.deinit();
-    highest_slot_meta.received = 20;
+        const r = try findSlotsToClean(&reader, 0, 100);
+        try std.testing.expectEqual(false, r.should_clean);
+        try std.testing.expectEqual(0, r.total_shreds);
+        try std.testing.expectEqual(0, r.highest_slot_to_purge);
+        var data_shred = try ledger.shred.DataShred.zeroedForTest(allocator);
+        defer data_shred.deinit();
+        {
+            var write_batch = try db.initWriteBatch();
+            defer write_batch.deinit();
+            for (0..1000) |i| {
+                try write_batch.put(ledger.schema.schema.data_shred, .{ 19, i }, data_shred.payload);
+            }
+            try db.commit(&write_batch);
+        }
+        // When implementation is not rocksdb, we do not need to flush memtable to disk to be able to assert.
+        if (build_options.blockstore_db != .rocksdb) {
```

Comment:
Instead of separate scopes and repeating the same assertions multiple times, what do you think about just adding this before the assertions?

```zig
    if (build_options.blockstore_db == .rocksdb) {
        db.deinit();
        db = try TestDB.reuseBlockstore(@src());
        reader.db = db;
    }
```